src/DGtal/math/Profile.h

```diff
@@ -0,0 +1,384 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file Profile.h
+ * @author Bertrand Kerautret (\c kerautre@loria.fr )
+ * LORIA (CNRS, UMR 7503), University of Nancy, France
+ * @Jacques-Olivier Lachaud
+ *
+ * @date 2015/11/08
+ *
+ * Header file for module Profile.cpp
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(Profile_RECURSES)
+#error Recursive header files inclusion detected in Profile.h
+#else // defined(Profile_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define Profile_RECURSES
+
+#if !defined Profile_h
+/** Prevents repeated inclusion of headers. */
+#define Profile_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <vector>
+#include "DGtal/math/Statistic.h"
+
+#include "DGtal/base/Common.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // class Profile
+  /**
+   * Description of class 'Profile' <p> \brief Aim: This class can be
+   * used to represent a profile (PX, PY) defined from an input set of
+   * samples (Xi, Yi). From all the samples (Xk, Yk) having the same
+   * value Xk, the associated value PY is computed (by default) by the
+   * mean of the values Yk. Note that other definitions can be used
+   * (MAX, MIN or MEDIAN).
+   *
```

Comment:
maybe you should say that internally, each sample abscissa is an instance of DGtal::Statistic